api/raft.cc

```diff
@@ -74,12 +74,33 @@ void set_raft(http_context&, httpd::routes& r, sharded<service::raft_group_regis
             return json_return_type(srv.current_leader().to_sstring());
         });
     });
+    r::read_barrier.set(r, [&raft_gr] (std::unique_ptr<http::request> req) -> future<json_return_type> {
+        auto srv = std::invoke([&] {
+            if (!req->query_parameters.contains("group_id")) {
+                return raft_gr.local().group0_with_timeouts();
+            }
+            raft::group_id gid{utils::UUID{req->get_query_param("group_id")}};
+            return raft_gr.local().get_server_with_timeouts(gid);
+        });
+        auto timeout = std::invoke([timeout_str = req->get_query_param("timeout")] {
+            if (timeout_str.empty()) {
+                return ::service::raft_timeout{};
+            }
+            auto dur = std::stoll(timeout_str);
+            if (dur <= 0) {
+                throw bad_param_exception{"Timeout must be a positive number."};
+            }
+            return ::service::raft_timeout{.value = lowres_clock::now() + std::chrono::seconds{dur}};
+        });
+        co_await srv.read_barrier(nullptr, timeout);
```

Comment:
@kbr-scylla Doesn't group0 server live only on shard 0?

Reply:
Yes

Reply:
So this whole thing needs to happen on the correct shard for the requested group (and if no group requested -- on shard 0 for group 0)

For example similar API, `trigger_snapshot`, uses this:
```cpp
        co_await raft_gr.invoke_on_all([gid, timeout_dur, &found_srv] (service::raft_group_registry& raft_gr) -> future<> {
            auto* srv = raft_gr.find_server(gid);
            if (!srv) {
                co_return;
            }

```

by calling `invoke_on_all` for the `raft_group_registry` service, all shards but one (the one containing the group) will do nothing

Reply:
fixed the code to run on the correct shards